apps/grader-host/src/index.ts

```diff
@@ -324,8 +325,15 @@ async function initFiles(context: Context) {
     logger.info('Unzipping files');
     await execa('tar', ['-xf', jobArchiveFile.path, '-C', jobDirectory.path]);
 
+    // TODO: AFAICT this is to handle entrypoints that start with /grade/ or
+    // equivalent (e.g., in /grade/serverFilesCourse or /grade/tests). If that
+    // is the case, this code should only run if the entrypoint is indeed in
+    // that directory.
```

Comment:
@nwalters512 I believe the correct handling here is to check if the entrypoint (or more precisely the first token of the entrypoint) starts with `/grade`. Can you double-check if this makes sense?

Reply:
That sounds right, yeah.

Reply:
Note that we probably want to ensure that we do something sensible for an entrypoint like `/grade/../../../other-thing` (that is, guard against directory traversals).

Reply:
Done in 4b6bccca9dc87cba8c664ff70211fff739e0e3b0.